src/app/api/view-settings/viewSettings.controller.ts

```diff
@@ -0,0 +1,24 @@
+import { NextRequest, NextResponse } from 'next/server'
+import authenticate from '@api/core/utils/authenticate'
+import { ViewSettingsService } from '@api/view-settings/viewSettings.service'
+import { CreateViewSettingsSchema } from '@/types/dto/viewSettings.dto'
+
+export const getViewSetting = async (req: NextRequest) => {
+  const user = await authenticate(req)
+
+  const viewSettingsService = new ViewSettingsService(user)
+  const viewSetting = await viewSettingsService.getViewSettingsForUser()
+
+  return NextResponse.json({ viewSetting })
```

Comment:
why do we wrap the viewSetting in `{}`. Just pass this back directly

Reply:
This is supposed to be a response building design pattern so we could add more keys to the response if required without changing the response schema drastically @makeitraina 

E.g. I can easily accommodate changes on the API consumer if the response changed to `NextResponse.json({ viewSetting, newkey: value })` with this pattern - while I would have to do a lot more changes if I just returned `NextResponse.json(viewSetting)`

I see how it could be redundant for our use case though, so I will be removing it for now

cc @aatbip

Reply:
lets have convention of sending `data` in responses in that case

```
non paginated response: { data: {} }
paginated response: { data: {}, meta: {}, links: {} }
```